src/macros.jl

```diff
@@ -446,19 +362,12 @@ macro infinite_parameter(args...)
     end
 
     # finalize the code and register the parameter name if needed
-    if is_anon
-        macro_code = creation_code
-    else
-        macro_code = _macro_assign_and_return(_error, creation_code, name,
-                                              esc_model)
-    end
-    return _finalize_macro(_error, esc_model, macro_code, __source__)
+    return _finalize_macro(error_fn, model, creation_code, __source__, name)
 end
```

Comment:
This one is still fairly complicated. But better. And it isn't obvious how to improve what remains.

Reply:
Ya I never expected this one to get much better